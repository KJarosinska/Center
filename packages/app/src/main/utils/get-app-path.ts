/**
 * Copyright (c) Mudita sp. z o.o. All rights reserved.
 * For licensing, see https://github.com/mudita/mudita-center/blob/master/LICENSE.md
 */

import path from "path"
import { name } from "../../../package.json"

export type AppType = "main" | "renderer"
export type AppResolver = () => {
  type: AppType
  app: { getPath: (arg0: string) => string }
}

export const resolve: AppResolver = () => {
  const isRenderer = require("is-electron-renderer")
  const app = isRenderer
    ? require("electron").remote.app
    : require("electron").app
  const type = isRenderer ? "renderer" : "main"
  return {
    app,
    type,
  }
}

const getAppPath = (): string => {
  try {
    return path.join(resolve().app.getPath("appData"), name)
  } catch {
    return ""
  }
}

export default getAppPath
