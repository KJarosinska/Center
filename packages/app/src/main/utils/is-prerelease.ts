/**
 * Copyright (c) Mudita sp. z o.o. All rights reserved.
 * For licensing, see https://github.com/mudita/mudita-center/blob/master/LICENSE.md
 */

export const isPrerelease = (tagName: string): boolean => {
  return tagName.replace(/(release-[\d]*\.[\d]*\.[\d]*)/i, "") !== ""
}
