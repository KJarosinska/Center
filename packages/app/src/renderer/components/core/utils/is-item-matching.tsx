export type IsItemMatching = (item: any, searchString: string) => boolean
export const isItemValueMatching: IsItemMatching = (
  itemValue: string,
  search
) => itemValue.toLowerCase().includes(search.toLowerCase())
