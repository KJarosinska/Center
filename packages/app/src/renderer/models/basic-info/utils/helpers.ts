import { SimCard } from "Renderer/models/basic-info/basic-info.typings"

export const getActiveNetworkFromSim = (simCards: SimCard[] = []) => {
  const activeSimCard = simCards.filter((simCard) => simCard.active)
  if (activeSimCard.length === 0) {
    return "No connection"
  }
  return activeSimCard[0].network
}
export const getActiveNetworkLevelFromSim = (simCards: SimCard[] = []) => {
  const activeSimCard = simCards.filter((simCard) => simCard.active)
  if (activeSimCard.length === 0) {
    return "No connection"
  }
  return activeSimCard[0].networkLevel
}
