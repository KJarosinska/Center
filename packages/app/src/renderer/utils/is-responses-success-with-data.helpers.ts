/**
 * Copyright (c) Mudita sp. z o.o. All rights reserved.
 * For licensing, see https://github.com/mudita/mudita-center/blob/master/LICENSE.md
 */

import RequestResponse, {
  RequestResponseStatus,
} from "App/main/functions/request-response.interface"

interface SuccessRequestResponse<DataType = undefined>
  extends RequestResponse<DataType> {
  status: RequestResponseStatus.Ok
  data: DataType
}

export const isResponsesSuccessWithData = (
  responses: RequestResponse<any>[]
): responses is SuccessRequestResponse<any>[] => {
  return responses.every(
    ({ status, data }) =>
      status === RequestResponseStatus.Ok && data !== undefined
  )
}
