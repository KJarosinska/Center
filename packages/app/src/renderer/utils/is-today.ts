import moment, { MomentInput } from "moment"

export const isToday = (date: MomentInput) =>
  moment(date).isSame(Date.now(), "days")
