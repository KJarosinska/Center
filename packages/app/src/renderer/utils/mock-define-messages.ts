import { defineMessages } from "react-intl"

export const mockDefineMessages = (id = "view.name.news") => {
  const messages = defineMessages({
    exampleMessage: { id },
  })
  return messages.exampleMessage
}
