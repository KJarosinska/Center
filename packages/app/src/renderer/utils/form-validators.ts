/**
 * Copyright (c) Mudita sp. z o.o. All rights reserved.
 * For licensing, see https://github.com/mudita/mudita-center/blob/master/LICENSE.md
 */

import { RegisterOptions } from "react-hook-form/dist/types"
import { intl } from "Renderer/utils/intl"

export const phoneNumberRegexp = /^(\+?)(\d(\s?\d)+)$/im

export const passwordRegexp =
  /^(?=.*\d)(?=.*[!@#$%^&*])(?=.*[a-z])(?=.*[A-Z]).{8,}$/

export const primaryPhoneNumberValidator = (
  fields: Record<any, any>
): RegisterOptions => ({
  minLength: {
    value: 5,
    message: intl.formatMessage(
      { id: "component.formErrorTooShort" },
      { minLength: 5 }
    ),
  },
  maxLength: {
    value: 15,
    message: intl.formatMessage(
      { id: "component.formErrorTooLong" },
      { maxLength: 15 }
    ),
  },
  pattern: {
    value: phoneNumberRegexp,
    message: intl.formatMessage({
      id: "component.formErrorNumbersAndSpacesOnly",
    }),
  },
  validate: (value): string | undefined => {
    if (value.length > 0 && value === fields.secondaryPhoneNumber) {
      return intl.formatMessage({
        id: "component.formErrorNumberUnique",
      })
    }

    return
  },
})

export const secondaryPhoneNumberValidator = (
  fields: Record<any, any>
): RegisterOptions => ({
  minLength: {
    value: 5,
    message: intl.formatMessage(
      { id: "component.formErrorTooShort" },
      { minLength: 5 }
    ),
  },
  maxLength: {
    value: 15,
    message: intl.formatMessage(
      { id: "component.formErrorTooLong" },
      { maxLength: 15 }
    ),
  },
  pattern: {
    value: phoneNumberRegexp,
    message: intl.formatMessage({
      id: "component.formErrorNumbersAndSpacesOnly",
    }),
  },
  validate: (value): string | undefined => {
    if (value.length > 0 && value === fields.primaryPhoneNumber) {
      return intl.formatMessage({
        id: "component.formErrorNumberUnique",
      })
    }

    return
  },
})

export const emailValidator: RegisterOptions = {
  pattern: {
    value:
      /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/,
    message: intl.formatMessage({ id: "component.formErrorInvalidEmail" }),
  },
  required: intl.formatMessage({ id: "component.formErrorRequiredEmail" }),
}

export const backupSecretKeyValidator = {
  required: {
    value: true,
    message: intl.formatMessage({
      id: "module.overview.backupSecretKeyRequired",
    }),
  },
  minLength: {
    value: 8,
    message: intl.formatMessage(
      { id: "component.formErrorTooShort" },
      { minLength: 8 }
    ),
  },
  // pattern: {
  //   value: passwordRegexp,
  //   message: intl.formatMessage({
  //     id: "component.formErrorNumbersAndSpacesOnly",
  //   }),
  // },
  validate: (value: string): string | undefined => {
    if (passwordRegexp.test(value)) {
      return
    } else if (value.search(/(?=.*[A-Z])/)) {
      return intl.formatMessage({
        id: "module.overview.backupSecretKeyUppercase",
      })
    } else if (value.search(/(?=.*\d)/)) {
      return intl.formatMessage({ id: "module.overview.backupSecretKeyNumber" })
    } else if (value.search(/(?=.*[!@#$^%&? "])/)) {
      return intl.formatMessage({
        id: "module.overview.backupSecretKeySpecialCharacter",
      })
    } else if (value.search(/(?=.*[a-z])/)) {
      return intl.formatMessage({
        id: "module.overview.backupSecretKeyLowercase",
      })
    }

    return intl.formatMessage({
      id: "module.overview.backupSecretKeyConfirmationDoesntMatch",
    })
  },
}
