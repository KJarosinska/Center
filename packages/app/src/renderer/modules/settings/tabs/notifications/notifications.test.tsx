/**
 * Copyright (c) Mudita sp. z o.o. All rights reserved.
 * For licensing, see https://github.com/mudita/mudita-center/blob/master/LICENSE.md
 */

import { renderWithThemeAndIntl } from "Renderer/utils/render-with-theme-and-intl"
import React from "react"
import NotificationsUI from "Renderer/modules/settings/components/notifications-ui.component"

test("matches snapshot", () => {
  const { container } = renderWithThemeAndIntl(
    <NotificationsUI
      appIncomingCalls={false}
      appIncomingMessages={false}
      appLowBattery={false}
      appOsUpdates={false}
    />
  )
  expect(container).toMatchSnapshot()
})
