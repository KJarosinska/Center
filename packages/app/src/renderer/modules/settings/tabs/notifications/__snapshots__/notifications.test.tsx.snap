// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
.c2 {
  margin: 0;
  font-size: 1.4rem;
  line-height: 1.57;
  -webkit-letter-spacing: 0.04rem;
  -moz-letter-spacing: 0.04rem;
  -ms-letter-spacing: 0.04rem;
  letter-spacing: 0.04rem;
  font-weight: 300;
  color: #000000;
}

.c6 {
  margin: 0;
  font-size: 1.6rem;
  line-height: 1.5;
  -webkit-letter-spacing: 0.02rem;
  -moz-letter-spacing: 0.02rem;
  -ms-letter-spacing: 0.02rem;
  letter-spacing: 0.02rem;
  color: #000000;
}

.c15 {
  margin: 0;
  font-size: 1.2rem;
  line-height: 1.67;
  font-weight: 400;
  -webkit-letter-spacing: 0.1rem;
  -moz-letter-spacing: 0.1rem;
  -ms-letter-spacing: 0.1rem;
  letter-spacing: 0.1rem;
  text-transform: uppercase;
  color: #000000;
}

.c7 {
  grid-area: Name;
  -webkit-align-self: end;
  -ms-flex-item-align: end;
  align-self: end;
}

.c9 {
  grid-area: Actions;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 9rem;
}

.c12 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  background: none;
  -webkit-transition: background 250ms ease-in-out, color 250ms ease-in-out, border 250ms ease-in-out;
  transition: background 250ms ease-in-out, color 250ms ease-in-out, border 250ms ease-in-out;
  cursor: pointer;
  outline: none;
  box-sizing: border-box;
  width: 17.6rem;
  height: 4rem;
  color: #ffffff;
  border-radius: 0.5rem;
  background: #3b3f42;
  border: 0.1rem solid #3b3f42;
}

.c12:hover {
  background: #232426;
}

.c12 g {
  fill: #ffffff;
}

.c17 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  background: none;
  -webkit-transition: background 250ms ease-in-out, color 250ms ease-in-out, border 250ms ease-in-out;
  transition: background 250ms ease-in-out, color 250ms ease-in-out, border 250ms ease-in-out;
  cursor: pointer;
  outline: none;
  box-sizing: border-box;
  width: 17.6rem;
  height: 4rem;
  border-radius: 0.5rem;
  border: 0.1rem solid #000000;
}

.c17 g {
  fill: #000000;
}

.c17:hover {
  background: #f4f5f6;
}

.c16 {
  white-space: nowrap;
}

.c10 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  position: relative;
}

.c13 {
  --radius: 0.5rem;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  z-index: 1;
  pointer-events: all;
  z-index: 3;
}

.c13:disabled {
  pointer-events: none;
}

.c13:hover {
  z-index: 2;
}

.c13:not(:first-of-type) {
  border-radius: 0;
  margin-left: -0.1rem;
}

.c13:first-of-type:not(:last-of-type) {
  border-radius: var(--radius) 0 0 var(--radius);
}

.c13:last-of-type:not(:first-of-type) {
  border-radius: 0 var(--radius) var(--radius) 0;
}

.c18 {
  --radius: 0.5rem;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  z-index: 1;
  pointer-events: all;
}

.c18:disabled {
  pointer-events: none;
}

.c18:hover {
  z-index: 2;
}

.c18:not(:hover) {
  border: 0.1rem solid #d2d6db;
  color: #6a6a6a;
}

.c18:not(:hover) g {
  fill: #f4f5f6;
}

.c18:not(:first-of-type) {
  border-radius: 0;
  margin-left: -0.1rem;
}

.c18:first-of-type:not(:last-of-type) {
  border-radius: var(--radius) 0 0 var(--radius);
}

.c18:last-of-type:not(:first-of-type) {
  border-radius: 0 var(--radius) var(--radius) 0;
}

.c11 {
  margin-right: 3.2rem;
}

.c14 {
  padding: 0 3.6rem;
  width: 8.8rem;
}

.c4 {
  display: grid;
  box-sizing: border-box;
  grid-template-areas: "Checkbox Actions";
  grid-template-columns: 1fr 15rem;
  border-bottom: solid 0.1rem #d2d6db;
  height: 7.2rem;
  max-height: 7.2rem;
}

.c5 {
  grid-area: Checkbox;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c8 {
  margin-left: 3.2rem;
  margin-bottom: 0;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
}

.c1 {
  border-bottom: solid 0.1rem #d2d6db;
}

.c3 {
  margin-left: 3.2rem;
  margin-bottom: 3.2rem;
}

.c0 {
  padding-top: 3.2rem;
}

<div>
  <section
    class="c0"
  >
    <div
      class="c1"
    >
      <p
        class="c2 c3"
        color="primary"
      >
        [value] module.settings.notificationsDescription
      </p>
    </div>
    <div
      class="c4"
    >
      <div
        class="c5"
      >
        <p
          class="c6 c7 c8"
          color="primary"
        >
          [value] module.settings.notificationsIncomingCallsNotificationsLabel
        </p>
      </div>
      <div
        class="c9"
      >
        <section
          class="c10 c11"
        >
          <button
            class="c12 c13 c14"
            data-testid="toggler-active"
            size="2"
            type="button"
          >
            <p
              class="c15 c16"
              color="primary"
            >
              [value] module.settings.offLabel
            </p>
          </button>
          <button
            class="c17 c18 c14"
            data-testid="toggler-inactive"
            size="2"
            type="button"
          >
            <p
              class="c15 c16"
              color="primary"
            >
              [value] module.settings.onLabel
            </p>
          </button>
        </section>
      </div>
    </div>
    <div
      class="c4"
    >
      <div
        class="c5"
      >
        <p
          class="c6 c7 c8"
          color="primary"
        >
          [value] module.settings.notificationsIncomingMessagesNotifications
        </p>
      </div>
      <div
        class="c9"
      >
        <section
          class="c10 c11"
        >
          <button
            class="c12 c13 c14"
            data-testid="toggler-active"
            size="2"
            type="button"
          >
            <p
              class="c15 c16"
              color="primary"
            >
              [value] module.settings.offLabel
            </p>
          </button>
          <button
            class="c17 c18 c14"
            data-testid="toggler-inactive"
            size="2"
            type="button"
          >
            <p
              class="c15 c16"
              color="primary"
            >
              [value] module.settings.onLabel
            </p>
          </button>
        </section>
      </div>
    </div>
    <div
      class="c4"
    >
      <div
        class="c5"
      >
        <p
          class="c6 c7 c8"
          color="primary"
        >
          [value] module.settings.notificationsLowBatteryNotifications
        </p>
      </div>
      <div
        class="c9"
      >
        <section
          class="c10 c11"
        >
          <button
            class="c12 c13 c14"
            data-testid="toggler-active"
            size="2"
            type="button"
          >
            <p
              class="c15 c16"
              color="primary"
            >
              [value] module.settings.offLabel
            </p>
          </button>
          <button
            class="c17 c18 c14"
            data-testid="toggler-inactive"
            size="2"
            type="button"
          >
            <p
              class="c15 c16"
              color="primary"
            >
              [value] module.settings.onLabel
            </p>
          </button>
        </section>
      </div>
    </div>
    <div
      class="c4"
    >
      <div
        class="c5"
      >
        <p
          class="c6 c7 c8"
          color="primary"
        >
          [value] module.settings.notificationsPureOsUpdatesNotifications
        </p>
      </div>
      <div
        class="c9"
      >
        <section
          class="c10 c11"
        >
          <button
            class="c12 c13 c14"
            data-testid="toggler-active"
            size="2"
            type="button"
          >
            <p
              class="c15 c16"
              color="primary"
            >
              [value] module.settings.offLabel
            </p>
          </button>
          <button
            class="c17 c18 c14"
            data-testid="toggler-inactive"
            size="2"
            type="button"
          >
            <p
              class="c15 c16"
              color="primary"
            >
              [value] module.settings.onLabel
            </p>
          </button>
        </section>
      </div>
    </div>
  </section>
</div>
`;
